### PR TITLE
[math] Port to Win64 (replace Long_t by Longptr_t + pointer formatting

### PR DESCRIPTION
**Pull request by bellenot**

### CODE REVIEW

File: math/matrix/src/TMatrixTBase.cxx [hunk 1]
```diff
@@ -622,8 +622,8 @@ Element TMatrixTBase<Element>::Max() const
 template<class Element>
 void TMatrixTBase<Element>::Draw(Option_t *option)
 {
-   gROOT->ProcessLine(Form("THistPainter::PaintSpecialObjects((TObject*)0x%lx,\"%s\");",
-                           (ULong_t)this, option));
+   gROOT->ProcessLine(Form("THistPainter::PaintSpecialObjects((TObject*)0x%zx,\"%s\");",
+                           (size_t)this, option));
```
Comment (Axel-Naumann) [inline math/matrix/src/TMatrixTBase.cxx:626]: As "this" is a pointer, can we not use `Longptr_t` here, too?

**Review by lmoneta — APPROVED** [review]
Looks fine, but I do not understand the issue. 
Why one cannot use Long_t on Windows, Long_t is a ROOT defined type, and why not  defining it correctly on Windows ? Why not using directly size_t ? 
Maybe some description in the PR could be also helpful

File: math/matrix/src/TMatrixTBase.cxx (hunk 1 above)
Comment (bellenot) [reply math/matrix/src/TMatrixTBase.cxx:626]: Well, nope, the `%zx` formatting requires `size_t` and is fully cross architecture. And using `%lx` or `%llx` will then conflict, depending on the platform. The best and more portable way is really this one (it took me some time to find the best way to fix type casting errors and warnings)

**Comment by bellenot** [comment]
@moneta `Long_t ` is 32 bit on Win64 and cannot hold a pointer. And replacing its definition with another type break all the code (e.g. template arguments, methods overload, ...). That's the only possible way not to break other platforms, other architectures, and stay backward compatible

**Comment by bellenot** [comment]
@lmoneta see also https://github.com/root-project/root/commit/c82701984fc457baa3ad6c616eb37da3bd59a0d6

File: math/matrix/src/TMatrixTBase.cxx (hunk 1 above)
Comment (Axel-Naumann) [reply math/matrix/src/TMatrixTBase.cxx:626]: Right, while `std::uint64_t` is a good, conservative target type for a pointer cast, it's all but obvious which format specifier to use... Thanks!

**Review by Axel-Naumann — APPROVED** [review]

**Pull request merged**